Lib/test/test_io.py

```diff
@@ -1726,6 +1737,16 @@ def bad_write(b):
         self.assertRaises(OSError, b.close) # exception not swallowed
         self.assertTrue(b.closed)
 
+    def test_slow_close_from_thread(self):
+        rawio = self.SlowFlushRawIO()
+        bufio = self.tp(rawio, 8)
+        t = threading.Thread(target=bufio.close)
+        t.start()
+        time.sleep(0.1) # Long enough that t is sleeping in bufio.close()
```

Comment:
You could probably use an `Event` that you would set in `SlowFlushRawIO.flush()` instead.